src/main.rs

```diff
@@ -315,7 +315,7 @@ fn generate_shadow() -> io::Result<()> {
 
     for line in reader.lines() {
         if let Some((username, _)) = line?.split_once(':') {
-            writeln!(writer, "{}:!:::::::", username)?;
+            writeln!(writer, "{}::::::::", username)?;
```

Comment:
just to be sure: is it OK to do that with "system" users? e.g. the ones having `nologin` or `false` for the shell.

Reply:
You raised a valid point... even with localhost only another user might be able to ssh into the guest, potentially having access to data that was only visible to the user that started the vng instance.

So this change seems to have security implications and it's probably safer to stick with the public key authentication.

I'll think more about this, but I'm more inclined to drop this change.

Reply:
> even with localhost only another user might be able to ssh into the guest, potentially having access to data that was only visible to the user that started the vng instance.

Good point. But then, do we not have the same issue with the VSOCK option? :-/

Reply:
In any cases, we should restrict the SSH connection to localhost only by default I think.

Reply:
Do we have any kind of permission restriction with vsock? If not we should mention it in the README or the help.

Maybe we should consider vsock a quick (but potentially unsafe) way to connect to the vng instance and if you need better security use ssh.

And yes, we definitely need to restrict access to localhost only.

Reply:
> Do we have any kind of permission restriction with vsock? If not we should mention it in the README or the help.

It looks like we have some restrictions with the current code "by accident": when starting the VM, an empty script file will be created (by default `/tmp/virtme-console/2222.sh`) and only the current user can modify it. So we can connect to the vsock, but nothing will happen apart from executing an empty script. Looks enough, no?

If we want something more secure, it looks like QEmu can take a Unix socket in argument in order not to have the CID accessible to everybody on the machine (via `/dev/vhost-vsock`), but I didn't really find a lot of documentation about that, except this [Rust VMM project](https://github.com/rust-vmm/vhost-device/blob/main/vhost-device-vsock/README.md) that seems to be using a Unix socket. Do you think it is worth investigating? It feels like what we have for the moment is enough.

Now regarding SSH, maybe warning the user in the README/help menu is enough?